src/ml_tooling/transformers/rare_feature.py

```diff
@@ -0,0 +1,41 @@
+import pandas as pd
+from typing import Any, Union
+from ml_tooling.metrics.utils import _is_percent
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class RareFeatureEncoder(BaseEstimator, TransformerMixin):
+    def __init__(self, threshold: Union[int, float] = 0.2, fill_rare: Any = "Rare"):
+        """
+        Replaces categories with a specified value, if they occur less often than
+        the provided threshold.
+
+        Parameters
+        ----------
+        threshold: int, float
+            Threshold of a categorie minimum occurrence.
```

Comment:
Maybe a bit more sentencey?

"Sets the threshold for when a value is considered rare. Any value which occurs less than (or equal?) to the threshold will be replaced with `fill_rare`. If threshold is a float, it will be considered a percentage and if it is an int, `threshold` will be considered the minimum number of observations"